### PR TITLE
[SYCL][ESIMD] Make logs architecture more flexible

### PR DESCRIPTION
**Pull request by yuriykoch**
Removes tight coupling of the test case description and error message, so that
we could to use any error messages with no need of manual message construction.
Any log message specific details are moved to the upper level, to the message
construction itself, because log messages are entry points for test failure
analysys.
Also a syntax sugar for trace and debug messages added.

Signed-off-by: Kochetkov, Yuriy <yuriyx.kochetkov@intel.com>

### CODE REVIEW

File: SYCL/ESIMD/api/functional/ctors/common.hpp [hunk 1]
```diff
@@ -14,36 +14,26 @@
...
 
-template <typename DataT, int NumElems, typename ContextT>
+template <int NumElems, typename ContextT>
 class TestDescription : public ITestDescription {
 public:
-  TestDescription(size_t index, DataT retrieved_val, DataT expected_val,
-                  const std::string &data_type)
-      : m_data_type(data_type), m_retrieved_val(retrieved_val),
-        m_expected_val(expected_val), m_index(index) {}
+  TestDescription(const std::string &data_type) : m_data_type(data_type) {}
```
Comment (vasilytric) [inline SYCL/ESIMD/api/functional/ctors/common.hpp]: Why you do not provide a backward compatibility? Without the backward compatibility new tests will be failed on CI when this PR will be merged

please leave the previous version of the `TestDescription` constructor without any changed in `to_string` method

Comment (yuriykoch) [reply SYCL/ESIMD/api/functional/ctors/common.hpp]: Would you mind to take a look on a backward compatibility suggestion? This PR removes unused `DataT` template parameter, so we have a difference in template parameters also.

File: SYCL/ESIMD/api/functional/ctors/common.hpp [hunk 2]
```diff
@@ -14,20 +14,44 @@
...
+  const std::string m_data_type;
+};
+
+#else
+
+// Deprecated, use TestDescription<NumElems, ContextT> for new tests instead
+//
+// TODO: Remove deprecated TestDescription from all tests
+template <typename DataT, int NumElems, typename ContextT>
+class [[deprecated]] TestDescription : public ITestDescription {
```
Comment (yuriykoch) [inline SYCL/ESIMD/api/functional/ctors/common.hpp:47]: @vasilytric This attribute would trigger a compiler warning for every template instantiation by-design. Let me know if that's not needed from your point of view.

Comment (vasilytric) [reply SYCL/ESIMD/api/functional/ctors/common.hpp:47]: I think that this attribute will be useful

**Comment by yuriykoch** [comment]
@sndmitriev Would you mind to take a look? It's a generic code, so I'm going to support this branch in case any other test is merged - still there is no any dependency in merge order with other PRs.

**Comment by yuriykoch** [comment]
Left the `operator_logical_not` test untouched, with deprecated logs enabled. Looks like the log messages update there could trigger code refactoring out of scope of this PR due to equal error messages for different logic.
Tagging @vasilytric for awareness

**Comment by yuriykoch** [comment]
@sndmitriev Friendly ping

**Review by sndmitriev — APPROVED** [review]

**Comment by yuriykoch** [comment]
@kbobrovs Would you mind to merge this PR?

**Pull request merged**